src/core/renderer_gl/renderer_gl.cpp

```diff
@@ -38,18 +38,165 @@ const char* fragmentShader = R"(
 	uniform uint u_alphaControl;
 	uniform uint u_textureConfig;
 
+	// TEV uniforms
+	uniform uint u_textureEnvSource[6];
+	uniform uint u_textureEnvOperand[6];
+	uniform uint u_textureEnvCombiner[6];
+	uniform vec4 u_textureEnvColor[6];
+	uniform uint u_textureEnvScale[6];
+	uniform uint u_textureEnvUpdateBuffer;
+	uniform vec4 u_textureEnvBufferColor;
+
 	// Depth control uniforms
 	uniform float u_depthScale;
 	uniform float u_depthOffset;
 	uniform bool u_depthmapEnable;
 
 	uniform sampler2D u_tex0;
 
+	// TODO: figure out the color that is returned to the TEVs when reading from a disabled texture slot.
+	vec4 tev_texture_sources[4] = vec4[](vec4(0.0), vec4(0.0), vec4(0.0), vec4(0.0));
+
+	// TODO: figure out the correct "initial" value returned to TEV0 when reading the "previous" source.
```

Comment:
I believe black will do for now. Citra handles various GPU UB (eg drawing a texture from an invalid paddr) by rendering it as black and it seems to work good enough.

Reply:
Yeah, this would probably require extensive hardware tests and the results could be weird.